pkg/component/tasks.go

```diff
@@ -29,15 +29,17 @@ import (
 // TaskFunc is the task function.
 type TaskFunc func(context.Context) error
 
-func (f TaskFunc) Execute(ctx context.Context) (err error) {
+func (f TaskFunc) Execute(ctx context.Context, logger log.Interface) (err error) {
 	defer func() {
 		if p := recover(); p != nil {
-			fmt.Fprintf(os.Stderr, "%s\n%s", p, debug.Stack())
+			fmt.Fprintln(os.Stderr, p)
+			os.Stderr.Write(debug.Stack())
 			if pErr, ok := p.(error); ok {
 				err = errTaskRecovered.WithCause(pErr)
 			} else {
 				err = errTaskRecovered.WithAttributes("panic", p)
 			}
+			logger.WithError(err).Error("Task panicked")
```

Comment:
We're logging this here in order to allow `Error` to capture the correct stack trace. Also the `logger` is passed as the one in `ctx` does not contain the invocations / task name.